src/lua/timer.rs

```diff
@@ -0,0 +1,255 @@
+use rlua::{UserData, UserDataMethods};
+
+use super::{
+    backend::Backend,
+    blight::Blight,
+    constants::{BACKEND, TIMED_CALLBACK_TABLE, TIMED_CALLBACK_TABLE_CORE},
+};
+use crate::event::Event;
+use chrono::Duration;
+use std::error::Error;
+use std::sync::Arc;
+
+pub struct Timer {
+    next_id: u32,
+}
+
+impl Timer {
+    pub fn new() -> Self {
+        Self { next_id: 0 }
+    }
+
+    fn next_index(&mut self) -> u32 {
```

Comment:
You can remove this and extrapolate the next index from rlua::Table.raw_len() instead.